include/coro/ring_buffer.hpp

```diff
@@ -10,6 +10,20 @@
 
 namespace coro
 {
+namespace rb
+{
+enum class produce_result
```

Comment:
I moved these since using them was sort of impossible/really annoying with the template parameter on the coro::ring_buffer class.